File: TwoThreadTest/cpp/TwoThreadTest.cpp
#include "SimpleThread.h"
#include <thread>

/**
 * Runs two instances of SimpleThread
 */
int main() {
    SimpleThread thread1("Hi");
    SimpleThread thread2("Ho");

    std::thread hi(&SimpleThread::run, &thread1);
    std::thread ho(&SimpleThread::run, &thread2);

    hi.join();
    ho.join();

    return 0;
}

File: TwoThreadTest/java/TwoThreadTest.java
package java;
/**
 * Runs two instances of SimpleThread.
 */
public class TwoThreadTest {
    public static void main(String[] args) {
        SimpleThread thread1 = new SimpleThread("Hi");
        SimpleThread thread2 = new SimpleThread("Ho");

        thread1.start();
        thread2.start();
    }
}
